pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -426,6 +426,11 @@ def resolve_training_type_plugin(self, training_type: TrainingTypePlugin) -> Tra
         if hasattr(training_type, 'num_nodes') and getattr(training_type, 'num_nodes') is None:
             training_type.num_nodes = self.num_nodes
 
+        # Automatically set sync_batchnorm.
+        # Useful for custom plugins.
+        if hasattr(training_type, 'sync_batchnorm'):
```

Comment:
Shouldn't this be `not hasattr`? Or check that it's `None` before overwriting

Reply:
It should be `hasattr` but you're right, there is a check for None missing here

Reply:
It gets set to `False` by default, so the attribute is already there, and therefore the `None` check won't work. 

For built-in plugins, it's already being set here https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/connectors/accelerator_connector.py#L400, so then these lines of codes will just be a noop. It will only be useful for custom plugins.

Reply:
Okay, I get you.

@justusschock Do you think it would be safer to make it default to `None` to differentiate between built-in and custom plugins? Is overwriting the property potentially dangerous?

Reply:
I don't think it is set by default. Only if the class is not yet instantiated (which is not the case here). 

So for your custom plugin instance, the attribute must be present and have a default of None. The trainer, however, should not have a default of None but the current one. Thus you need to check here if the attribute is None. (In fact after instantiation, `None` should be required to have the trainer control this).

@carmocca Overwriting this should be fine, since the synced_bn conversion happens only if the model is available (i.e. during `fit`, which is definitely after setting up plugins and accelerators

Reply:
@justusschock OK makes sense. I made the change to add the `None` check in the if statement.

Reply:
@amogkam Looks good to me, thanks :)